### PR TITLE
Chore: Add grafana-apiserver

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

This adds the k8s https://github.com/grafana/grafana-apiserver as a dependency and updates go.mod so everything plays nicely.

**Why do we need this feature?**

Keeping the go.mod in sync with a complex feature branch is difficult.  In this PR we will introduce the build dependencies, but will continue to iterate the faster changing/more experimental parts of the project in a feature branch

**Who is this feature for?**

### CODE REVIEW

File: pkg/services/k8s/apiserver/watchset.go
Comment (charandas) [inline pkg/services/k8s/apiserver/watchset.go:1]: Is watchset needed? Looks a lot like what we have been using from Tilt, but in Filestorage implementation.

File: pkg/tsdb/tempo/trace_transform.go [hunk 1]
```diff
@@ -205,75 +168,67 @@ func getAttributeVal(attr pdata.AttributeValue) interface{} {
...
+	return nil
+}
+
+func getTagFromTraceState(traceState pdata.TraceState) *KeyValue {
+	return nil
+}
 
-	return logs
+func spanEventsToLogs(events pdata.SpanEventSlice) []*TraceLog {
+	return nil
```
Comment (charandas) [inline pkg/tsdb/tempo/trace_transform.go:231]: Is this what you mean by commented code? Was this method emptied to make the build pass due to the otel library downgrade?

Comment (toddtreece) [reply pkg/tsdb/tempo/trace_transform.go:231]: yeah, i think we'll need to do something like this to reset those files and see if they build:

```
git checkout main pkg/tsdb/tempo pkg/infra/httpclient/httpclientprovider/tracing_middleware.go
```

**Comment by charandas** [comment]
Rebased from `main` post merging my otel related changes.

File: go.mod [hunk 2]
```diff
@@ -28,7 +40,7 @@ replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.
 
 require (
 	cloud.google.com/go/storage v1.28.1
-	cuelang.org/go v0.6.0-0.dev
+	cuelang.org/go v0.5.0
```
Comment (charandas) [inline go.mod:43]: Not sure where this is coming from.

**Review by adrapereira — COMMENTED** [review]
Tracing changes make sense 👍

File: pkg/modules/registry/registry_test.go [hunk 3]
```diff
@@ -9,31 +9,8 @@ import (
 
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/modules"
-	"github.com/grafana/grafana/pkg/server/backgroundsvcs"
 )
 
-func TestProvideRegistry(t *testing.T) {
```
Comment (toddtreece) [inline pkg/modules/registry/registry_test.go:15]: @wbrowne it seemed like this test was going to be too difficult to maintain as we add new dependencies to the registry. do you it's ok to rely on `TestNewRegistry`, or should we look into adding mocks for each dependency we add?

Comment (wbrowne) [reply pkg/modules/registry/registry_test.go:15]: Nah I think it's fine to kill this for now. An alternative is to just comment it out and when things are more set in stone, we can revisit and update/remove.

**Review by wbrowne — COMMENTED** [review]
Looks good AFAICT. Would appreciate the eyes of some backend platform folks on the changes to the `web` package though.

File: pkg/modules/modules.go [hunk 4]
```diff
@@ -149,3 +164,31 @@ func (m *service) RegisterInvisibleModule(name string, initFn func() (services.S
...
+		for parent, targets := range dependencyMap {
+			for _, t := range targets {
+				if t == GrafanaAPIServer && m.IsModuleEnabled(parent) {
+					return fmt.Errorf("feature flag %s is disabled, but target %s is enabled with dependency on %s", featuremgmt.FlagGrafanaAPIServer, parent, GrafanaAPIServer)
+				}
+			}
+		}
+	}
+
+	return nil
```
Comment (toddtreece) [inline pkg/modules/modules.go:193]: @ryantxu does this logic seem ok for handling the feature flag?

Comment (ryantxu) [reply pkg/modules/modules.go:193]: For this PR -- i think this is sufficient.  But I think we want to make a more clear relationship between targests and featureflags.  I am exploring that in a branch, but it is not complicated for this PR :)

**Review by ryantxu — APPROVED** [review]
i think we should merge with main asap -- and then improve in smaller follow on PRs.  The patterns for setting up these services needs to get improved before we widely adopt it

**Pull request merged**